exchanges/order/timeinforce.go

```diff
@@ -0,0 +1,125 @@
+package order
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// var error definitions
+var (
+	ErrInvalidTimeInForce     = errors.New("invalid time in force value provided")
+	ErrUnsupportedTimeInForce = errors.New("unsupported time in force value")
+)
+
+// TimeInForce enforces a standard for time-in-force values across the code base.
+type TimeInForce uint16
+
+// TimeInForce types
+const (
+	UnknownTIF     TimeInForce = 0
+	GoodTillCancel TimeInForce = 1 << iota
+	GoodTillDay
+	GoodTillTime
+	GoodTillCrossing
+	FillOrKill
+	ImmediateOrCancel
+	PostOnly
+
+	supportedTimeInForceFlag = GoodTillCancel | GoodTillDay | GoodTillTime | GoodTillCrossing | FillOrKill | ImmediateOrCancel | PostOnly
+)
+
+// Is checks to see if the enum contains the flag
+func (t TimeInForce) Is(in TimeInForce) bool {
+	return in != 0 && t&in == in
+}
+
+// StringToTimeInForce converts time in force string value to TimeInForce instance.
+func StringToTimeInForce(timeInForce string) (TimeInForce, error) {
+	var result TimeInForce
+	timeInForce = strings.ToUpper(timeInForce)
+	switch timeInForce {
+	case "IMMEDIATEORCANCEL", "IMMEDIATE_OR_CANCEL", ImmediateOrCancel.String():
+		result = ImmediateOrCancel
+	case "GOODTILLCANCEL", "GOODTILCANCEL", "GOOD_TIL_CANCELLED", "GOOD_TILL_CANCELLED", "GOOD_TILL_CANCELED", GoodTillCancel.String():
+		result = GoodTillCancel
+	case "GOODTILLDAY", GoodTillDay.String(), "GOOD_TIL_DAY", "GOOD_TILL_DAY":
+		result = GoodTillDay
+	case "GOODTILLTIME", "GOOD_TIL_TIME", GoodTillTime.String():
+		result = GoodTillTime
+	case "GOODTILLCROSSING", "GOOD_TIL_CROSSING", "GOOD TIL CROSSING", GoodTillCrossing.String(), "GOOD_TILL_CROSSING":
+		result = GoodTillCrossing
+	case "FILLORKILL", "FILL_OR_KILL", FillOrKill.String():
+		result = FillOrKill
+	case PostOnly.String(), "POC", "POST_ONLY", "PENDINGORCANCEL":
+		result = PostOnly
+	case "POST_ONLY_GOOD_TIL_CANCELLED":
+		result = GoodTillCancel | PostOnly
```

Comment:
🚧  POST_ONLY_GOOD_TIL_CANCELLED was a bitrrex thing.
It's now unused, so we should remove it.

🥃 I understand the argument that it's an argument of how we might do this, but it's implicit that we can union bit types. And we shouldn't keep examples of things we don't currently need.
( e.g. I have mothball/* branches with loads of types of subscription key types that became obsolete. I'll probably never need them again )

Feel free to seek a quorum from others if you still disagree. I'm never offended by overruled by consensus 🙂

[This thread](https://github.com/thrasher-corp/gocryptotrader/pull/1382#discussion_r2043684185) was migrated from the outdated file